signet-program/programs/signet/src/lib.rs

```diff
@@ -409,126 +726,178 @@ pub struct ReadRespond<'info> {
+/// by recovering the public key and comparing with the expected derived key.
 #[event]
 pub struct SignatureRespondedEvent {
+    /// Request identifier linking this response to the original request.
+    /// Computed as `keccak256(sender || payload || ...)` - see module docs.
     pub request_id: [u8; 32],
+    /// Address of the MPC responder (not cryptographically verified).
     pub responder: Pubkey,
+    /// ECDSA signature in affine point format.
     pub signature: Signature,
 }
 
-/**
- * @dev Emitted when a signature error is received.
- * @notice Any address can emit this event. Do not rely on it for business logic.
- * @param request_id The ID of the request. Must be calculated off-chain.
- * @param responder The address of the responder.
- * @param error The error message.
- */
+/// Emitted when signature generation fails via [`chain_signatures::respond_error`].
+///
+/// # Warning: Debugging Only
+///
+/// This event is **solely for debugging purposes**. Do not use in production
+/// or rely upon for any business logic.
+///
+/// # Event Type
+///
+/// Regular event (emitted via `emit!`)
+///
+/// # Security Warning
+///
+/// **Any address can emit this event.** Error events are not cryptographically
+/// verified and should never be trusted for business logic decisions.
 #[event]
 pub struct SignatureErrorEvent {
+    /// Request identifier of the failed request.
     pub request_id: [u8; 32],
+    /// Address of the MPC responder (not cryptographically verified).
     pub responder: Pubkey,
+    /// Human-readable error description.
     pub error: String,
 }
 
-/**
- * @dev Emitted when a read response is received.
- * @param request_id The ID of the request. Must be calculated off-chain.
- * @param responder The address of the responder.
- * @param serialized_output The serialized output.
- * @param signature The signature.
- */
+/// Emitted when the MPC network returns execution results for a bidirectional
+/// request via [`chain_signatures::respond_bidirectional`].
+///
+/// # Event Type
+///
+/// Regular event (emitted via `emit!`)
+///
+/// # Output Format
+///
+/// **Successful transactions:**
+/// - Contract call: Return value serialized per `respond_serialization_schema`
+/// - Simple transfer: Empty success indicator
+///
+/// **Failed transactions:**
+/// - Magic prefix `0xdeadbeef` followed by failure indicator
+///
+/// # Signature Verification
+///
+/// The signature is computed over `keccak256(request_id || serialized_output)` using
+/// the special derivation path `"solana response key"`. See module-level docs for
+/// verification procedure.
+///
+/// # Security Warning
+///
+/// **Any address can emit this event.** Clients **must** verify the signature
+/// before trusting the output.
 #[event]
 pub struct RespondBidirectionalEvent {
+    /// Original request identifier.
     pub request_id: [u8; 32],
+    /// Address of the MPC responder (not cryptographically verified).
```

Comment:
`(not photographically verified)` is a bit missleading here, since the responder is always correct, but the signature is not guaranteed to be valid